encodings/dict/src/compress.rs

```diff
@@ -187,6 +161,33 @@ where
     )
 }
 
+fn dict_values_validity(nullable: bool, len: usize) -> Validity {
+    if nullable {
+        Validity::Array(
+            SparseArray::try_new(
```

Comment:
I suspect it might be better to use RunEndBool array? because I optimized that one's `IntoCanonical`, whereas `canonicalize_sparse_bools` in SparseArray uses `Vec<bool>` (which should be fixed)

Reply:
(will also serialize smaller)

Reply:
My other pr fixes canonicalization of sparse bools